flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/AdaptiveExecutionPlanSchedulingContext.java

```diff
@@ -118,6 +119,14 @@ public int getPendingOperatorCount() {
         return adaptiveGraphManager.getPendingOperatorsCount();
     }
 
+    @Override
+    public String getJsonStreamGraph() {
+        return JsonPlanGenerator.generateJsonStreamGraph(
+                adaptiveGraphManager.getStreamGraphContext().getStreamGraph(),
+                adaptiveGraphManager.getStreamNodeIdsToJobVertexMap(),
```

Comment:
How about adding `getJsonStreamGraph` and `generateJsonStreamGraph` methods to the `AdaptiveGraphManager`? 
We can call `generateJsonStreamGraph` to trigger an update when the stream graph is updated or new job vertices are created. For example, in the `DefaultStreamGraphContext#modifyStreamEdge`, we could call `generateJsonStreamGraph` if the update result is true. Similarly, in the `AdaptiveGraphManager#onJobVertexFinished`, we also should call `generateJsonStreamGraph`.

Therefore, the `AdaptiveExecutionPlanSchedulingContext#getJsonStreamGraph` only needs to call AdaptiveGraphManager#getJsonStreamGraph.

With this update, we could benefit from the following aspects:

1. We do not need to update the `immutable stream graph` because the `AdaptiveGraphManager` can handle the stream graph.
2. We avoid frequently generating the JSON stream graph, as we only update it when the graph is updated.